### PR TITLE
Feature/agent1 adbist

### PR DESCRIPTION
**Pull request by abist**
Made a few changes, that we can decide to take or not. Here's the summary -
1. Removed loading all job histories in the jobs view page, and just query for the history of the job clicked on. It's because the perf was very bad, to the point that testing the history page was taking 2-3 mins. It's much better now.
2. Now all previous runs show in the left side in the history page. Clicking on any of those previous runs would change the overview details to match the scope of the clicked job.

Also, since we don't have all the fields from the mock-up, can we get a different list of properties to show when the history page is visited? I've changed a couple on my own, just to get an opinion.

### CODE REVIEW

File: extensions-modules/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "0.1.0",
   "description": "Shared modules for SQL Operations Studio extensions",
   "dependencies": {
-    "dataprotocol-client": "github:Microsoft/sqlops-dataprotocolclient#0.1.5",
+    "dataprotocol-client": "github:Microsoft/sqlops-dataprotocolclient#feature/agentDmp1",
```
Comment (kburtram) [inline extensions-modules/package.json]: please revert the change to this file

File: src/sql/parts/jobManagement/views/jobHistory.component.html [hunk 2]
```diff
@@ -98,31 +98,31 @@ <h1 class="job-heading">Jobs | {{agentJobInfo.name}} </h1>
...
-			{{agentJobInfo.lastRun}}
+			{{agentJobHistoryInfo?.runDate}}
 		</h1>
 		<table class="step-list">
 			<tr class="step-row">
 				<td height="30">
 					<h3>Status:</h3>
 				</td>
-				<td>
+				<td height="30">
```
Comment (kburtram) [inline src/sql/parts/jobManagement/views/jobHistory.component.html]: should we create a style for this?

File: src/sql/parts/jobManagement/views/jobHistory.component.ts [hunk 3]
```diff
@@ -111,22 +140,26 @@ export class JobHistoryComponent extends Disposable implements OnInit, OnDestroy
 		}
 	}
 
-	private jobAction(action: string): void {
+	private jobAction(action: string, jobName): void {
```
Comment (kburtram) [inline src/sql/parts/jobManagement/views/jobHistory.component.ts]: can you add a type to jobName?  In general I suggest we try to explicitly provide types when declaring variables.  But we should certainly do this for function parameters.

File: src/sql/sqlops.d.ts [hunk 4]
```diff
@@ -1061,13 +1061,13 @@ declare module 'sqlops' {
 	}
 
 	export interface AgentJobHistoryInfo {
-		instanceID: number;
+		instanceId: number;
```
Comment (kburtram) [inline src/sql/sqlops.d.ts]: any reason we're switching ID to Id?  In Typescript is the normal convention `ID`?  In any case, I don't think it's a good idea to mix and match the two casing conventions within a single interface.

I think I may have been using `Id` since that is the C# convention.  If you see that, please feel free to switch to the correct convention or note in code review.  Thanks!

File: src/sql/parts/jobManagement/views/jobHistory.component.html (hunk 2 above)
Comment (abist) [reply src/sql/parts/jobManagement/views/jobHistory.component.html]: cells have to be inline-styled

File: src/sql/sqlops.d.ts (hunk 4 above)
Comment (abist) [reply src/sql/sqlops.d.ts]: @kburtram Every single variable or field with `ID` in the `sqlops.d.ts` file is in Pascal case (`Id`). I'm assuming it's because that's the way we receive it from the service.
